src/cache/cacher.rs

```diff
@@ -57,6 +57,7 @@ pub trait Cacher: Send + Sync {
     /// failure.
     async fn cache_results(
         &mut self,
+        options: u8,
```

Comment:
```suggestion
```

I think you are going with the code based approach of feature options. While the issue expects that you have to do it according to the rust's conditional compiling method of feature based approach. I think I would suggest you to watch the above-mentioned video tutorial on how to write code with rust's conditional feature for this. :slightly_smiling_face: 

Also, Here are some other resources on how to write code with conditional compiled feature:

- https://stackoverflow.com/questions/27632660/how-do-i-use-conditional-compilation-with-cfg-and-cargo
- https://betterprogramming.pub/compile-time-feature-flags-in-rust-why-how-when-129aada7d1b3
- https://stackoverflow.com/questions/60287822/can-i-create-my-own-conditional-compilation-attributes

Which can help you better understand and write code in this aspect. :slightly_smiling_face:

Reply:
I see the importance of cfg but I may need more hints :/

Reply:
> I see the importance of cfg but I may need more hints :/

@spencerjibz, I think you may have a better idea on this. Would you like to provide some hints to evanyang1 ? :slightly_smiling_face: